### PR TITLE
Feature achievements

### PR DESCRIPTION
**Pull request by jessearens**
Still have to improve test methods and coverage, but would like to implement what I got so far.

<feature,testing

### CODE REVIEW

File: src/main/java/com/github/fishio/achievements/Achieve.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+import com.github.fishio.logging.Log;
+import com.github.fishio.logging.LogLevel;
+
+/**
+ * This class loads a set of Achievements and sets the rules that have to be met
+ * in order for the achievement to be obtained.
+ *
+ */
+public class Achieve {
```
Comment (Diocruel) [inline src/main/java/com/github/fishio/achievements/Achieve.java]: _Auto-generated comment_

```
Utility classes should not have a public or default constructor
```

CheckStyle error. Please fix this.

<layout,organisation

File: src/test/java/com/github/fishio/achievements/TestAchievement.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+	
+	/**
+	 * Tests the {@link Achievement#Achievement(String)} constructor.
+	 */
+	@Test
+	public void testAchievement2() {
+		Achievement achievetest2 = new Achievement("Testachievement");
+	}
+
```
Comment (Diocruel) [inline src/test/java/com/github/fishio/achievements/TestAchievement.java:61]: When testing if the constructor works you should do some kind of check that the constructor initialized correctly. 

https://blackboard.tudelft.nl/bbcswebdav/pid-2391002-dt-content-rid-7957602_2/courses/34547-141501/TI1206-Q1-15-ExtraTesten.pdf 

<testing

**Comment by Diocruel** [comment]
I think the things I noted should be fixed, this can be done in one or two commits. Please provide a bit more of a small walkthrough through the code next time, as observer patterns are new to many of us too. Lastly, in the walkthrough you should also specify what code you want us to be looking at, because  you already did a previous pull request were you already implemented some of the code here (from one branch to another). Telling us what code is already checked will save everybody time.

Good job though on implementing the observer design pattern and the test mostly look really good. Hope you can fix this today!

<testing,layout,organisation,feature,naming

**Comment by ChielBruin** [comment]
Two tests are failing:
testAchievementconstructor1_twoAchievements
testAchievementconstructor2_twoAchievements

so closing PR
<test

**Comment by jessearens** [comment]
Please mainly take a look at the last 6 commits, with exception of  5feb31e. Other commits have already been through a pull request from a subbranch into this branch.

This pull request adds the achievements from the already approved observer pattern for the properties. Next up would be to link the achievement screen with these achievements.

<test,feature,bug,review

**Comment by ChielBruin** [comment]
Looks good!
The progress is real!

**Comment by Diocruel** [comment]
I agree. Thanks for implementing all the fixes!

+1 

<feature,  testing, review

**Pull request merged**